src/main/java/app/model/agents/ACO/AcoAgent.java

```diff
@@ -0,0 +1,322 @@
+package app.model.agents.ACO;
+
+import app.controller.graphicsEngine.Ray;
+import app.controller.linAlg.Vector;
+import app.model.Move;
+import app.model.agents.AgentImp;
+import app.model.agents.Cells.GraphCell;
+import app.model.agents.MemoryGraph;
+import app.model.agents.Team;
+import app.model.agents.WallFollow.WfWorld;
+import lombok.Getter;
+import lombok.Setter;
+import org.jgrapht.graph.DefaultEdge;
+
+import java.util.*;
+
+public class AcoAgent extends AgentImp
+{
+    @Getter private static int acoAgentCount;
+    @Getter private static int acoMoveCount;
+
+    @Getter private final double maxPheromone = 2;
+    private final double epsilon = 0.3;
+    private static Random randomGenerator = new Random(1);
+    private final int[] cardinalAngles = {0, 90, 180, 270};
+
+    @Getter private ArrayList<Vector> possibleMovements = new ArrayList<>();
+    @Getter private Stack<Vector> visualDirectionsToExplore = new Stack<>();
+    @Getter private ArrayList<Vector> pheromoneDirections = new ArrayList<Vector>();
+    //TODO Discuss dynamic movement heuristic value
+    @Setter private double movementHeuristic = 0.95;
+    @Setter private Vector movementContinuity = new Vector();
+    //TODO Discuss variable visionDistance heuristic
+    @Getter @Setter private double visionDistance = 30.0;
+    @Getter @Setter private int distance = 20;
+    @Getter @Setter protected Move previousMove;
+    @Getter private HashMap<Integer, Vector> shortTermMemory = new HashMap<>();
```

Comment:
Is there a purposeful order to your variable declarations?  They seem to be randomly assigned. Perhaps lead with the `final static` vars, then  `static` and then the rest in a logical order to make it easier to parse.

Reply:
Will merge as it is, then take a separate branch to complete minor refactors so that reviews don't have to me completed again for this PR. Thanks!